packages/react/src/components/popover/useObserver.ts

```diff
@@ -0,0 +1,31 @@
+import { RefObject, useEffect, useRef } from 'react';
+
+/**
+ * Observes `targets` for intersection changes.
+ * @param targets `RefObject` of element to observe.
+ * @param onChange Called when the threshold changes. Either the element just became visible or the opposite.
+ * @param debounceTime Time in milliseconds to debounce events.
+ */
+export function useObserver<T extends Element>(
```

Comment:
Is this generic providing flexibility on the type element because all `HTMLElements` inherit from Element? This is a nice use of generics :)

Reply:
Correct. Observers require an `Element` only so we need not be more specific than that.

In this case the generic doesn't matter because we eat `T` but it would if the return type was T, for instance.
If we did not use the generic then, it'd squash the definition from the outside.

For example:
```ts
const buttonRef = useRef<HTMLButtonElement>(null);

const getCurrentWithGenerics = <T extends Element>(ref: RefObject<T>) =>
  ref.current;
const getCurrentWithPlainType = (ref: RefObject<Element>) => ref.current;

const noiceType = getCurrentWithGenerics(buttonRef);
// HTMLButtonElement | null

const sadType = getCurrentWithPlainType(buttonRef);
// Element | null
```